### PR TITLE
Add basic support for issue-config file defaults

### PR DESCRIPTION
**Pull request by kkaarreell**
Adds support for issue config file default settings. Such a settings is applied to individual actions defined under `issues` attribute.
Example:
```
defaults:
  assignee: null
  fields:
    "Pool Team": "my_great_team"
    "Story Points": 0
```
`defaults` are merged when defined in multiple (included) files.

### CODE REVIEW

File: newa/cli.py [hunk 1]
```diff
@@ -511,13 +521,35 @@ def _jira_fake_id_generator() -> Generator[str, int, None]:
...
+
             # Iterate over issue actions. Take one, if it's not possible to finish it,
             # put it back at the end of the queue.
             while issue_actions:
                 action = issue_actions.pop(0)
 
                 ctx.logger.info(f"Processing {action.id}")
 
+                # update action object with default attributes when not present
+                action = _update_action_with_defaults(action, config.defaults)
```
Comment (lukaszachy) [inline newa/cli.py:551]: Why is handling of defaults done here in the loop and not during `config = IssueConfig.from_yaml_with_include` call?

Comment (kkaarreell) [reply newa/cli.py:551]: Well, I guess I see that function for loading YAML, not doing semantic evaluation. Are you strongly against keeping it here?

Comment (lukaszachy) [reply newa/cli.py:551]: > Well, I guess I see that function for loading YAML, not doing semantic evaluation.

That function already extends YAML loader with 'include'  and isn't the 'default' closer to syntax? From YAML perspective it is just a default dictionary to update with whatever it reads from the rest of the file.

IMO 'issue_actions' should contain "final" classes before they reach 'while' loop.

Comment (kkaarreell) [reply newa/cli.py]: I have pushed an update, pls review.

**Review by lukaszachy — APPROVED** [review]
Works great, however more documentation will be appreciated - seems that if multiple layers of 'include' are used, the `defaults` is used from the top one (closest to the file used in `read_file()` call)

File: newa/cli.py (hunk 1 above)
Comment (lukaszachy) [reply newa/cli.py]: I don't see 'resolve' button so please mark this thread as resolved.

**Comment by kkaarreell** [comment]
> Works great, however more documentation will be appreciated - seems that if multiple layers of 'include' are used, the `defaults` is used from the top one (closest to the file used in `read_file()` call)

I see. I think that might be relevant for `include` processing in general. Ideally, users would avoid conflicts. I will test the behavior a bit more and either document it or modify it (if reasonably doable).

**Comment by kkaarreell** [comment]
> > Works great, however more documentation will be appreciated - seems that if multiple layers of 'include' are used, the `defaults` is used from the top one (closest to the file used in `read_file()` call)
> 
> I see. I think that might be relevant for `include` processing in general. Ideally, users would avoid conflicts. I will test the behavior a bit more and either document it or modify it (if reasonably doable).

I have pushed and update changing how the merging is done. Also, I have updated README with more notes on the topic.
I do also plan to add a test testing it.

File: README.md [hunk 2]
```diff
@@ -46,6 +46,74 @@ TESTING_FARM_API_TOKEN
...
+The only exceptions are are `issues` and `defaults` which are merged.
+To unset a value defined in an included file one can set the value to `null`.
+
+### project
+
+Defines Jira project to be used by NEWA.
+
+### transitions
+
+This is a mapping which tells NEWA which particular issue states (and resolutions) it should be using. This settings depends on a particular Jira project. It is also possible to specify resolution separated by a dot, e.g. `Closed.Obsolete`.
```
Comment (lukaszachy) [inline README.md:106]: Could you enumerate which transitions are required and optional?

**Review by lukaszachy — APPROVED** [review]

**Pull request merged**